vars/infra.groovy

```diff
@@ -179,7 +179,11 @@ Object runMaven(List<String> options, String jdk = '8', List<String> extraEnv =
   List<String> mvnOptions = ['--batch-mode', '--show-version', '--errors', '--no-transfer-progress']
   if (useArtifactCachingProxy) {
     withArtifactCachingProxy {
-      mvnOptions += "-s $env.MAVEN_SETTINGS"
+      // If an artifact caching proxy provider has been correctly configured,
+      // add the corresponding settings.xml to Maven options
```

Comment:
Post-merge review (nitpick since the PR does the job): what is the relationship between artifact caching provider configured and the undefined/defined generic variable `env.MAVEN_SETTINGS`?

As I see the patch here, it's a validation to ensure that the variable is used only if defined: is that correct?

Reply:
env.MAVEN_SETTINGS is defined if an artifact caching proxy has been configured (ie not skipped and not failing any health check)